common/src/jni/main/cpp/conscrypt/native_crypto.cc

```diff
@@ -8026,7 +8062,9 @@ static jlong NativeCrypto_SSL_set_timeout(JNIEnv* env, jclass, jlong ssl_address
     }
 
     // Convert to seconds
-    long timeout = millis / 1000;
+    static const jlong INT_MAX_AS_JLONG = static_cast<jlong>(INT_MAX);
+    uint32_t timeout = static_cast<uint32_t>(std::max(0,
```

Comment:
The AppVeyor build implies that this version fails to compile on Windows.  I don't see anything wrong with it, but the error is

C:\projects\conscrypt\common\src\jni\main\cpp\conscrypt\native_crypto.cc(8067): error C2589: '(': illegal token on right side of '::'
C:\projects\conscrypt\common\src\jni\main\cpp\conscrypt\native_crypto.cc(8067): error C2059: syntax error: '::'
C:\projects\conscrypt\common\src\jni\main\cpp\conscrypt\native_crypto.cc(8066): error C2143: syntax error: missing '(' before '::'

I notice we don't #include &lt;algorithm&gt;, which is where std::min and std::max are technically defined, so maybe try that?